mantis-control-plane/mantis-control-plane-server/src/main/java/io/mantisrx/master/resourcecluster/TaskExecutorState.java

```diff
@@ -253,4 +253,14 @@ TaskExecutorRegistration getRegistration() {
     protected CompletableFuture<TaskExecutorGateway> getGateway() {
         return this.gateway;
     }
+
+    protected CompletableFuture<TaskExecutorGateway> reconnect() {
+        this.gateway = rpcService.connect(registration.getTaskExecutorAddress(), TaskExecutorGateway.class)
+            .whenComplete((gateway, throwable) -> {
+                if (throwable != null) {
+                    log.error("Failed to connect to the gateway", throwable);
```

Comment:
consider warn level logging + re-throwing exception

Reply:
The exception is being thrown again. The function CompletionStage::whenComplete doesn't affect the future's outcome, it only serves as a side-effect function.